### DISCUSSION TITLE
Not understanding how to use map_from_entries

### DISCUSSION THREAD

**Discussion by EvanZ** [duckdb/duckdb#14234] [General]
So, to make a long story short I want to replicate something I have in Spark where I use `collect_list` followed by `map_from_entries` (in the PySpark API) to transform a list of maps into one large dictionary containing k,v pairs. For example:

I'm not having much luck using `map_from_entries` in DuckDB so far. It seems to be collecting all the keys into a "key" array and all the values into a "value" array inside one dict:

```
SELECT
        MAP_FROM_ENTRIES(
            ARRAY[
                STRUCT_PACK(key := 'key1', value := 'value1'),
                STRUCT_PACK(key := 'key2', value := 'value2')
            ]
        ) AS test_map;
```

gives `{'key': ['key1', 'key2'], 'value': ['value1', ...` but what I'm looking for is `{'key1':'value1', 'key2':'value2',....}`

What am I missing? Or is there a bug here?

**Comment by soerenwolfers** [comment]
Works for me:

```sql
SELECT
        MAP_FROM_ENTRIES(
            ARRAY[
                STRUCT_PACK(key := 'key1', value := 'value1'),
                STRUCT_PACK(key := 'key2', value := 'value2')
            ]
        ) AS test_map;
```
```
┌────────────────────────────┐
│          test_map          │
│   map(varchar, varchar)    │
├────────────────────────────┤
│ {key1=value1, key2=value2} │
└────────────────────────────┘
```

on duckdb 1.1.1. 

You should file an issue if you don't get this on an upgraded version.

**Comment by Tishj** [comment]
Sounds like something fixed by https://github.com/duckdb/duckdb/pull/12734

Unless your key is not hashable

**Reply by EvanZ** [reply]
Yep that fixed it!
